homeassistant/components/local_ip/__init__.py

```diff
@@ -1,13 +1,12 @@
 """Get the local IP address of the Home Assistant instance."""
 import voluptuous as vol
 
-from homeassistant import config_entries
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_NAME
 from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 
-DOMAIN = "local_ip"
-PLATFORM = "sensor"
+from .const import DOMAIN, PLATFORM
 
 CONFIG_SCHEMA = vol.Schema(
     {DOMAIN: vol.Schema({vol.Optional(CONF_NAME, default=DOMAIN): cv.string})},
```

Comment:
I'm kinda configured, with the latest changes, the whole CONF_NAME can be removed from the schema now right?

Reply:
I agree, but wouldn't that need to be [after the first release cycle ](https://github.com/home-assistant/core/pull/33483#discussion_r401987546)?

Reply:
We can deprecate it for users that have configured via config.yaml. We can't remove it outright since the config yaml is validated with this also for existing config entries.

Reply:
Use `cp.deprecated`.